collections-arrays/CompareByteArraysInCsharp/CompareByteArraysInCsharp/CompareByteArrays.cs

```diff
@@ -0,0 +1,147 @@
+﻿using BenchmarkDotNet.Attributes;
+using System.Collections;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+namespace CompareByteArraysInCsharp
+{
+    [MemoryDiagnoser]
+    [Orderer(BenchmarkDotNet.Order.SummaryOrderPolicy.FastestToSlowest)]
+    [RankColumn]
+    public class CompareByteArrays
+    {
+        private readonly Random _rand;
+        private readonly byte[] _byteArray;  
+
+        public CompareByteArrays()
+        {
+            _rand = new Random();
+            _byteArray = GenerateRandomArray(10000000);
+        }
+        public IEnumerable<object[]> SampleByteArray()
```

Comment:
The current benchmark setup is in a sense invalid as you have two cases of the same size, one being the worst case scenario where the arrays are equal, and the other being two unequal arrays. Since the unequal case is created randomly, the benchmark will vary greatly and very possible could find a difference on the very first character of the array.
I could be wrong on this, but my gut feeling is a better way to benchmark would be to show a worst case as you have with one of the cases, and then have a second case where the two are the same except for changing one value in the middle. If you want to also have a case where the first element is different so you are dealing with the overhead into the method and a quick return, then that could be a third test case. For those I would add a label or something so it is clear in the results which case is which.